### ISSUE TITLE
automatically "ostree-ize" existing containers

### ISSUE THREAD

**Issue by cgwalters** [containers/bootc#845]
A toplevel goal of this project is to support upgrading existing ostree systems in place, so we need ostree support.

However, we also want to further hide ostree and push it into the background. There's multiple interlocking aspects to this.

The biggest one here is to avoid having to carry any "ostree stuff" in the base images by default.

This issue may mostly get fixed by https://github.com/ostreedev/ostree-rs-ext/pull/674 but I'm leaving this open here to track other fixes that may be needed (e.g. https://github.com/containers/bootc/pull/844 was required so far) and also to link to our roadmap.

**Comment by jmpolom** [comment]
Is a conceivable solution here to support or develop a "backend" that does not depend on ostree? Look to elemental-toolkit and kairos for inspiration along these lines. What is ostree providing specifically?

**Comment by cgwalters** [comment]
Yes, https://github.com/containers/bootc/issues/20 is related to that as is the ongoing composefs work.
But a toplevel goal of this project is [seamless upgrades for existing ostree users](https://containers.github.io/bootc/relationships.html#relationship-with-fedora-coreos-and-silverblue-etc).

**Comment by cgwalters** [comment]
Moved https://github.com/containers/bootc/pull/887 here.

But yes what I'm thinking here is we drop the requirement for having `sysroot` and `ostree -> sysroot/ostree` and just auto-create them if missing (and error if they're not what we expect).

I think if we do those two things it'd really push ostree into a much less visible/required state.
